### ISSUE TITLE
Parallel indexing with partial

### ISSUE THREAD

**Issue by alienxp03** [ankane/searchkick#1298]
At the moment, we are using [queue strategy](https://github.com/ankane/searchkick#queuing) to reindex our dataset. If I understand this correct, we have to implement the queue ourselves and manually call the `Searchkick::ProcessQueueJob` job to run the queued data. 

Previously, we were using `async` strategy. Once of the nice perks of using Searchkick is that we can do [partial reindex](https://github.com/ankane/searchkick#partial-reindexing). However, now that we have moved to `queue` strategy, does it mean that we have to write our own implementation for partial reindex. 

In short, does it mean that if we are using `queue` strategy, partial reindex won't work?

**Comment by alienxp03** [comment]
Upon checking the source code, looks like it's not supported by default

https://github.com/ankane/searchkick/blob/df57db37916ce67825a9879305c08b97ab7e9aff/lib/searchkick/record_indexer.rb#L18-L21

**Comment by LemonAndroid** [comment]
what is not supported by default?

**Comment by alienxp03** [comment]
If I use `queue` mode. Updated my reply above.

**Comment by LemonAndroid** [comment]
ok

**Comment by ankane** [comment]
Hey @alienxp03, as you found in the source code, it's not supported. This can probably be accomplished by having a separate queue for each partial method, but not sure it's worth the added complexity right now.
